### PR TITLE
163518692 User Can Like Article

### PR DESCRIPTION
**Pull request by abayo-luc**
#### What does this PR do?
Let authenticated user like an article or dislike an arleady liked article
#### Description of Task to be completed?
- Write tests for article likes
- The article like and dislike
- Fetch article with likes
#### How should this be manually tested?
- Clone the repo and  check out `ft-article-likes-dislike-163518692` branch
- Follow the instructions in `ReadMe`
- Using any `API client tool *POSTMAN*`,
- Create the user and login to get the ```POST /api/v1/users``` and ```POST /api/v1/users/login```
- Create an article using ```POST /api/v1/articles``` and get the ```slug```
- Access `POST /api/v1/articles/:slug/likes` endpoint for liking an article
- Access ```POST /api/v1/articles/:slug/dislikes``` endpoint for disliking the article
- Make sure to pass the token under `request headers`. It should be {`Authorization`: `Bearer <access-token>`} KEY, VALUE PAIR
- Access ```GET /api/v1/articles/:slug/likes``` to fetch article with its likes
#### Any background context you want to provide?
Trying to like article twice, cancel the first like. 
#### What are the relevant pivotal tracker stories?
#163518692

### CODE REVIEW

File: controllers/articlesController.js [hunk 1]
```diff
@@ -56,4 +56,46 @@ export default class PostController {
       return res.status(500).send({ message: error, status: 500 });
     }
   }
+
+  /**
```
Comment (Musigwa-zz) [inline controllers/articlesController.js]: Good description, please do the same in the rest of your files to keep consistency

File: controllers/articleLikesController.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+/**
+ *ArticleLikesController
+ *
+ * @class
+ */
+
+export default class ArticleLikesController {
+  /**
+   *
```
Comment (Musigwa-zz) [inline controllers/articleLikesController.js]: Please remove the unwanted space to keep consistency.

File: controllers/articleLikesController.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        });
+      }
+      await ArticleLike.destroy({
+        where: {
+          userId: req.user.id,
+          articleId: article.id
+        }
+      });
+
+      return res.status(202).json({ message: "Successfully disliked" });
```
Comment (Musigwa-zz) [inline controllers/articleLikesController.js]: @abayo-luc, can state the reason for using **`202`** status code?
According to the following ==> `https://httpstatuses.com/202`, **`202`** means the request has been accepted for processing, but the processing has not been completed.

File: tests/integration/articleLikes.test.js [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+            if (err) {
+              return done(err);
+            }
+            expect(response.status).eql(201);
+            expect(response.body).to.an("object");
+            expect(response.body).to.have.property("message");
+            expect(response.body.message).to.be.eql("Successfully liked");
+            done();
+          });
+      });
```
Comment (Musigwa-zz) [inline tests/integration/articleLikes.test.js:51]: Can you please try to keep a single space between the test suits to keep consistency?

**Review by Musigwa-zz — CHANGES_REQUESTED** [review]
The work actually looks great, but please consider going through the files and change JsDoc, spacing to meet the team's consistency.

**Comment by Musigwa-zz** [comment]
> Some files could not be reviewed due to errors:
> 
> Oops! Something went wrong! :(
> ESLint: 4.19.1.
> ESLint couldn't find the plugin "eslint-plugin-jsdoc". This can happen for a couple different reasons:
> 1. If ESLint is installed globally, then make sure eslint-plugin-jsdoc is also installed globally. A globally-installed ESLint cannot find a locally-installed plugin.
> 2. If ESLint is installed locally, then it's likely that the plugin isn't installed correctly. Try reinstalling by running the following:
>     npm i eslint-plugin-jsdoc@latest --save-dev
> If you still can't figure out the problem, please stop by https://gitter.im/eslint/eslint to chat with the team.

**Comment by Musigwa-zz** [comment]
abayo-luc, as reported by CodeClimate, the overall coverage was reduced by (-4%). Would you please consider covering all possible statements in your tests?
screen shot 2019-02-21 at 09 37 49
screen shot 2019-02-21 at 09 34 31 Your branch's coverage.

screen shot 2019-02-21 at 10 54 25

 The develop branch's coverage.

**Review by Musigwa-zz — APPROVED** [review]
Looks great after testing it locally.

File: .eslintrc [hunk 5]
```diff
@@ -1,45 +1,49 @@
 {
   "root": true,
   "extends": ["airbnb-base", "prettier"],
-  "plugins": ["prettier", "jsdoc", "eslint-plugin-jsdoc"],
```
Comment (espoirMur) [inline .eslintrc:4]: Why removing these plugins???

**Review by espoirMur — CHANGES_REQUESTED** [review]

**Review by espoirMur — APPROVED** [review]

**Pull request merged**